cdc/sink/producer/kafka/kafka.go

```diff
@@ -457,18 +457,20 @@ func topicPreProcess(topic string, protocol codec.Protocol, config *Config, sara
 
 	// when try to create the topic, we don't know how to set the `max.message.bytes` for the topic.
 	// Kafka would create the topic with broker's `message.max.bytes`,
-	// we have to make sure it's not greater than `max-message-bytes`
-	brokerMessageMaxBytes, err := getBrokerMessageMaxBytes(admin)
-	if err != nil {
-		log.Warn("TiCDC cannot find `message.max.bytes` from broker's configuration")
-		return errors.Trace(err)
-	}
+	// we have to make sure it's not greater than `max-message-bytes` for the default open protocol & craft protocol.
+	if protocol == codec.ProtocolDefault || protocol == codec.ProtocolCraft {
```

Comment:
Is this change related to https://github.com/pingcap/ticdc/pull/3192?

Reply:
Yes. I was thinking about should we check `max-message-bytes` < `message.max.bytes` for protocol which not batching messages, such as Canal-JSON, and I think it's unnecessary, because most of the time, the Canal-JSON message should be successfully sent, and once meet size too large problem, it's unrelated to `max-message-bytes`, so only check for open protocol and craft.